maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/ValidationService.java

```diff
@@ -102,6 +105,16 @@ public Boolean isUserActionValid(UserValidationDTO request) {
         return true;
     }
 
+    public boolean isUserAuthorisedToEditField(UserSummaryDTO userSummaryDTO, RestrictedField restrictedField, Action requestedAction) {
+        if (userSummaryDTO == null || userSummaryDTO.getRoleDataItem() == null) {
+            return false;
+        }
+
+        return userSummaryDTO.getRoleDataItem().stream().anyMatch(i -> i.getDataItem().equals(restrictedField.getField())
+            && i.getEnabled().equals("Y")
+            && ((createActions.contains(requestedAction) && "Y".equals(i.getInsertAllowed())) || (
+            updateActions.contains(requestedAction) && "Y".equals(i.getUpdateAllowed()))));
```

Comment:
Minor, but can we reformat this so it's a bit easier to read?

I'm not sure the `UserAction` is relevant here, the method just needs to check whether inserting or updating is allowed as we already have the `isUserActionValid` method for this.

I realise you're relying on the pre-existing `RoleDataItemDTO` which has `insertAllowed` and `updateAllowed` as Strings, but I think we should refactor to use booleans. I also think we can just have a single field since there are no instances in the DB where separate permissions are set for inserts and updates - It looks like the developers who originally implemented this over-engineered the solution a little.

However, happy to leave this for now if you wouldn't mind creating a Jira ticket?

Reply:
Happy to go either way on this. If I change the string to a boolean (which as you say, it should have really been to begin with) I suspect I'll need to add some mapping in somewhere unless we also change the data in the database.

Reply:
I think the change would need to happen in the Court Data API. Let's leave it for now and spin up a ticket.

Reply:
The `Action` has now been removed, which simplifies the check fairly substantially. Let us know if this looks good or if you spot any further scope for improvement.

Makes sense about the over-engineering, you're right that we don't have (at least not in dev or uat currently, have not checked prod) any instances where a field can either only be created (x)or updated.

Reply:
[Task created for updating the Court Data API](https://dsdmoj.atlassian.net/jira/software/projects/LCAM/boards/881/backlog?selectedIssue=LCAM-1491).